Refresh.GameServer/Importing/Importer.cs

```diff
@@ -181,39 +181,29 @@ private bool IsMip(Span<byte> rawData)
         return true;
     }
 
-    protected GameAssetType DetermineAssetType(Span<byte> data, TokenPlatform? tokenPlatform)
+    protected (GameAssetType, GameSerializationMethod) DetermineAssetType(Span<byte> data, TokenPlatform? tokenPlatform)
     {
-        // LBP assets
-        if (MatchesMagic(data, "TEX "u8)) return GameAssetType.Texture;
-        if (MatchesMagic(data, "GTF "u8)) return GameAssetType.GameDataTexture;
-        if (MatchesMagic(data, "MATb"u8)) return GameAssetType.Material;
-        if (MatchesMagic(data, "PLNb"u8)) return GameAssetType.Plan;
-        if (MatchesMagic(data, "LVLb"u8)) return GameAssetType.Level;
-        if (MatchesMagic(data, "CHKb"u8)) return GameAssetType.LevelChunk;
-        if (MatchesMagic(data, "GMTb"u8)) return GameAssetType.GfxMaterial;
-        if (MatchesMagic(data, "MSHb"u8)) return GameAssetType.Mesh;
-        if (MatchesMagic(data, "PALb"u8)) return GameAssetType.Palette;
-        if (MatchesMagic(data, "FSHb"u8)) return GameAssetType.Script;
-        if (MatchesMagic(data, "RECb"u8)) return GameAssetType.MoveRecording;
-        if (MatchesMagic(data, "VOPb"u8)) return GameAssetType.VoiceRecording;
-        if (MatchesMagic(data, "PTGb"u8)) return GameAssetType.Painting;
-        if (MatchesMagic(data, "PRFb"u8)) return GameAssetType.SyncedProfile;
-        if (MatchesMagic(data, "MATT"u8)) return GameAssetType.GriefSongState;
-        if (MatchesMagic(data, "SSPt"u8)) return GameAssetType.SoftPhysicsSettings;
-        if (MatchesMagic(data, "BEVb"u8)) return GameAssetType.Bevel;
-        if (MatchesMagic(data, "ANMb"u8)) return GameAssetType.Animation;
+        GameAssetType? lbpAssetType = GameAssetTypeExtensions.LbpMagicToGameAssetType(data.Slice(0, 3));
+        
+        if (lbpAssetType != null)
+        {
+            return (lbpAssetType.Value, (GameSerializationMethod)data[3]);
```

Comment:
This cast to `GameSerializationMethod` doesn't feel safe?

Reply:
It is safe, the table above only stores assets which follow LBP's standard serialization, and all assets which do that always will all have one of a known set of serialization methods

Reply:
Mostly worried about forged data. For example we might check against `Unknown` to see if it's valid, but the real value might be something that isn't in the bounds of the enum.

I would feel safer with a unit test proving this. Otherwise we should check against `Enum.IsDefined`.